### ISSUE TITLE
CoreDNS custom domains not working

### ISSUE THREAD

**Issue by SuperMarioo** [Azure/aks-engine#2146]
**Describe the bug**

I have followed this documentation to add custom domain

https://github.com/MicrosoftDocs/azure-docs/blob/master/articles/aks/coredns-custom.md

```apiVersion: v1
kind: ConfigMap
metadata:
  name: coredns-custom
  namespace: kube-system
data:
 mycompany.com.server: |
    mycompany.com:53 {
        errors
        cache 30
        proxy . 192.11.0.1  
    }
```
Pods are unable to resolve custom domain .
curl: (6) Could not resolve host:

**Steps To Reproduce**

kubectl apply -f corednsms.yaml
kubectl delete pod --namespace kube-system --label k8s-app=kube-dns

Exec  to pod and curl http URL with custom domain

**Expected behavior**

custom domain should be resolved 

**AKS Engine version**

v0.41.0

**Kubernetes version**

1.15.4

**Additional context**

CoreDNS-1.6.2 version 

Logs from CoreDNS:

[WARNING] No files matching import glob pattern: conf.d/Corefile*
[WARNING] No files matching import glob pattern: conf.d/Corefile*
[WARNING] No files matching import glob pattern: conf.d/Corefile*

**Comment by devigned** [comment]
Hi @SuperMarioo, thx for reaching out. Let's see if we can get this resolved.

Per coredns, I don't think there is anything special going on here. That is to say, we don't have any special Azure configuration happening here, so the main docs for coredns might be more helpful to you: https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/.

Is you DNS server reachable at `192.11.0.1`?
```
$ nslookup microsoft.com 192.11.0.1
;; connection timed out; no servers could be reached
```
This also might help: https://stackoverflow.com/questions/55612141/resolve-custom-dns-in-kubernetes-cluster-aks

**Comment by SuperMarioo** [comment]
Hi @devigned , thx for reply . 

DNS Server is reachable 

I have tried your links but I'm still unable to resolve custom DNS

**Comment by jjburr** [comment]
if you look at your CoreDNS config does it get overwritten? 

If so you will have to switch reconcile with ensureexists on /etc/kubernetes/addons/coredns.yml otherwise it will keep getting reconciled with the default coredns settings.

**Comment by jackfrancis** [comment]
Fixed in #2288
